compiler/src/main/java/com/dylibso/chicory/compiler/DirectoryCache.java

```diff
@@ -0,0 +1,197 @@
+package com.dylibso.chicory.compiler;
+
+import static java.nio.file.StandardCopyOption.ATOMIC_MOVE;
+
+import java.io.IOException;
+import java.nio.file.FileAlreadyExistsException;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.regex.Pattern;
+
+/**
+ * Disk-backed sharded directory cache.
+ */
+public class DirectoryCache {
```

Comment:
I'm a bit reluctant to roll out and, eventually, maintain our own version of a disk based cache.
I'm worried that we will waste time debugging issues with filesystems etc. (like on Windows etc.).

The alternative approach would be to define the interfaces in the core modules, spin a new Maven sub-module that will depend on, e.g., `ehcache` and have a minimal implementation based on it.

wdyt?

Reply:
I'll refactor that out.